### PR TITLE
Start tuning cmath functions

### PR DESCRIPTION
**Pull request by ckormanyos**
Now that most `<cmath>` functions are present, the purpose of this PR is to tune them, check/extend ranges and try to hit more non-covered lines.

### CODE REVIEW

**Comment by ckormanyos** [comment]
Hi Matt (@mborland) I'm gonna merge this one lickety split. I've got a goal to get to %99\\%$ coverage soon but not quite on this one.

Based on experiences from other projects, it takes about a few months and another thousand lines of test code to reach $100\\%$, which I'd like to put on the table as a reachable goal. Let's party.

**Comment by mborland** [comment]
> Hi Matt (@mborland) I'm gonna merge this one lickety split. I've got a goal to get to $99\\%$ coverage soon but not quite on this one.
> 
> 
> 
> Based on experiences from other projects, it takes about a few months and another thousand lines of test code to reach $100\\%$, which I'd like to put on the table as a reachable goal. Let's party.

Sounds good to me. For the new fast types I've been recycling the same test code so hopefully it doesn't drop the coverage too much.

File: include/boost/decimal/detail/cmath/exp.hpp [hunk 1]
```diff
@@ -43,20 +43,13 @@ constexpr auto exp_impl(T x) noexcept
...
-            else
-            {
-                result = x;
-            }
+            result = (signbit(x) ? zero : std::numeric_limits<T>::infinity());
         }
         else if (fpc == FP_NAN)
         {
-            result = x;
+            result = std::numeric_limits<T>::quiet_NaN();
```
Comment (mborland) [inline include/boost/decimal/detail/cmath/exp.hpp:50]: I think this should stay as `result = x` so we return the correct type of NAN from the input.

File: include/boost/decimal/detail/cmath/exp.hpp [hunk 2]
```diff
@@ -43,20 +43,13 @@ constexpr auto exp_impl(T x) noexcept
...
         }
-    }
+    } // LCOV_EXCL_LINE
```
Comment (mborland) [inline include/boost/decimal/detail/cmath/exp.hpp:52]: Is this needed?

**Comment by ckormanyos** [comment]
Well I made a few efforts to get better on funcs and also got a few lines coveredon 128/256 bit backends. I did not quite finish everything on the (non-published) TODO list. But it's headed in the right direction. So when cycles green, I'll push this thing.

Funcs are getting rather stable nowdays. Good stuff.

**Pull request merged**